substrate/frame/nfts/src/tests.rs

```diff
@@ -614,8 +614,13 @@ fn transfer_owner_should_work() {
 			Nfts::transfer_ownership(RuntimeOrigin::signed(account(1)), 0, account(2)),
 			Error::<Test>::Unaccepted
 		);
+		assert_eq!(System::consumers(&account(2)), 0);
+
 		assert_ok!(Nfts::set_accept_ownership(RuntimeOrigin::signed(account(2)), Some(0)));
+		assert_eq!(System::consumers(&account(2)), 1);
+
 		assert_ok!(Nfts::transfer_ownership(RuntimeOrigin::signed(account(1)), 0, account(2)));
+		assert_eq!(System::consumers(&account(2)), 1); // one consumer is added due to deposit repatriation
```

Comment:
It looks unchanged

Reply:
1) `set_accept_ownership` increases the number of consumers
2) `transfer_collection` will need to decrease that number, but it additionally repatriates the deposit, which adds one consumer. So, without that fix, the new owner will have 2 consumers added instead of just one on collection transfer